src/dataset.py

```diff
@@ -1,184 +1,123 @@
-
-            single_sample[b] = values
-
-        s2_data = torch.from_numpy(single_sample.values).float()
 
-        # -------------------------------------------------------------
-        # Sentinel-1
-        s1bands = ["VV", "VH"]
-        single_sample = pd.DataFrame(index=[f"ts{tstep}" for tstep in range(12)])
-        for b in s1bands:
-            fts = [x for x in self.df.columns if b in x and "ts" in x]
-            values = row[fts].values.astype(float)
-            idx_valid = values != self._NODATAVALUE
-
-            # Conversion to dB
-            values[idx_valid] = 20 * np.log10(values[idx_valid]) - 83
-
-            # This is actually no data and should ideally been handled by
-            # the mask in both input and target. For now, put to 0
-            # at some point, assign another nodata value (e.g. -999)
-            values[~idx_valid] = 0
+        eo_data = np.zeros((cls.NUM_TIMESTEPS, len(BANDS)))
+        band_mapping = {
+            "OPTICAL-B02-ts{}-10m": "B2",
+            "OPTICAL-B03-ts{}-10m": "B3",
+            "OPTICAL-B04-ts{}-10m": "B4",
+            "OPTICAL-B05-ts{}-20m": "B5",
+            "OPTICAL-B06-ts{}-20m": "B6",
+            "OPTICAL-B07-ts{}-20m": "B7",
+            "OPTICAL-B08-ts{}-10m": "B8",
+            "OPTICAL-B8A-ts{}-20m": "B8A",
+            "OPTICAL-B11-ts{}-20m": "B11",
+            "OPTICAL-B12-ts{}-20m": "B12",
+            "SAR-VH-ts{}-20m": "VH",
+            "SAR-VV-ts{}-20m": "VV",
+            "METEO-precipitation_flux-ts{}-100m": "total_precipitation",
+            "METEO-temperature_mean-ts{}-100m": "temperature_2m",
+        }
+        for df_val, presto_val in band_mapping.items():
+            column_names = [df_val.format(t) for t in range(cls.NUM_TIMESTEPS)]
+            values = row[column_names].values.astype(float)
+            idx_valid = values != cls._NODATAVALUE
+            if presto_val in ["VV", "VH"]:
+                # convert to dB
+                values[idx_valid] = 20 * np.log10(values[idx_valid]) - 83
+            elif presto_val == "total_precipitation":
+                # scaling, and AgERA5 is in mm, Presto expects m
+                values[idx_valid] = values[idx_valid] / (100 * 1000.0)
+            elif presto_val == "temperature_2m":
+                # remove scaling
+                values[idx_valid] = values[idx_valid] / 100
+            eo_data[:, BANDS.index(presto_val)] = values
+        static_band_mapping = {"DEM-alt-20m": "elevation", "DEM-slo-20m": "slope"}
+        for df_val, presto_val in static_band_mapping.items():
+            eo_data[:, BANDS.index(presto_val)] = row[df_val]
+
+        return eo_data, latlon, month, row["LANDCOVER_LABEL"] == 11
 
-            single_sample[b] = values
+    def __getitem__(self, idx):
+        raise NotImplementedError
 
-        s1_data = torch.from_numpy(single_sample.values).float()
+    @classmethod
+    def normalize_and_mask(cls, eo: np.ndarray):
```

Comment:
Right now, we set them to `0`. 

In this part of the function, they remain `_NODATAVALUE` - then, in `normalize_and_mask` that value gets set to `0`. `normalize_and_mask` gets called in the classes which implement `getitem` (`WorldCerealMaskedDataset` and `WorldCerealLabelledDataset`).

I did it this way because I want us to be able to swap this out for a "True" mask next.

Reply:
Ah ok makes sense!